git/operations.go

```diff
@@ -276,35 +276,94 @@ func changed(ctx context.Context, workingDir, ref string, subPaths []string) ([]
 	return splitList(out.String()), nil
 }
 
-func execGitCmd(ctx context.Context, args []string, config gitCmdConfig) error {
-	if trace {
-		print("TRACE: git")
-		for _, arg := range args {
-			print(` "`, arg, `"`)
+// Whilst debugging or developing, you may wish to filter certain git commands out of the logs.
+func exemptedTraceGitCommand(command string) bool {
+	exemptedCommands := []string{
+		// To filter out a certain git subcommand add it here, e.g.:
+		// "config",
+	}
+	for _, exemptedCommand := range exemptedCommands {
+		if exemptedCommand == command {
+			return true
+		}
+	}
+	return false
+}
+
+// traceGitCommand returns a log line that can be useful when debugging and developing git activity
+func traceGitCommand(args []string, config gitCmdConfig, stdout string, stderr string) string {
+	if exemptedTraceGitCommand(args[0]) {
+		return ""
+	}
+
+	prepare := func(input string) string {
+		output := strings.Trim(input, "\x00")
+		scanner := bufio.NewScanner(strings.NewReader(output))
+		lines := []string{}
```

Comment:
Unless you know the length of a slice before you allocate it, usually you'd let it be a zero value since `append` and `len` treat those the same as a zero-length slice.

    var lines []string

Reply:
(immaterial if you replace all this with strings.Replace)

Reply:
thanks for mentioning.  great suggestion.  I'm going to use strings.Replace like you said but that's helpful information to have.